blockwork/transforms/transform.py

```diff
@@ -902,8 +902,27 @@ class TSerialTransform(TypedDict):
     ifaces: dict[str, TISerialAny]
 
 
-class TSerialTransformResult(TypedDict):
+@dataclass(frozen=True, kw_only=True)
+class TransformResult:
+    exit_code: int | None
     run_time: float
+    ident: str
+    _accepted: bool = field(init=False, default=False, compare=False, repr=False, hash=False)
+
+    def resolve(self):
+        "Resolve this result, raising an error if it has not been accepted"
+        if self._accepted or self.exit_code == 0:
+            return
+        self.reject(f"exit_code was `{self.exit_code}`")
+
+    def reject(self, details: str = "result was rejected"):
+        "Reject this result, raising an error"
+        raise RuntimeError(f"{self.ident} failed: {details}")
+
+    def accept(self):
+        "Mark this result as accepted"
+        object.__setattr__(self, "_accepted", True)
+        return True
```

Comment:
Note, need to use setattr here as dataclass is frozen (which we want so transforms can't accidentally fiddle with the exit_code)